hazelcast/src/main/java/com/hazelcast/cardinality/hyperloglog/impl/HyperLogLogEncType.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cardinality.hyperloglog.impl;
+
+import com.hazelcast.cardinality.hyperloglog.IHyperLogLog;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+
+import java.io.IOException;
+
+public enum HyperLogLogEncType {
+
+    SPARSE {
```

Comment:
Could you add some JavaDoc here to explain briefly the different types? I guess they differ in precision vs. memory usage (and maybe runtime).